roles/java/tasks/keystores.yml

```diff
@@ -28,6 +28,11 @@
         mode: "0400"
         dest: "{{ download_location }}"
       register: certcp
+    - name: Assign variable based on distribution
+      ansible.builtin.set_fact:
+        legacy: "-legacy"
+      when:
+        - (ansible_distribution == 'Rocky' or ansible_distribution == 'Redhat') and ansible_distribution_version is version('9', '>=')
```

Comment:
Isn't `ansible_os_family` set to `RedHat` with Rocky?
Also AFAIR `RedHat` needs a capital "H". Please double check.